### DISCUSSION TITLE
Multi-domain forwarding YARP setup in Azure hosting recommendations

### DISCUSSION THREAD

**Discussion by Piedone** [dotnet/yarp#1160] [Q&A] [answered]
Which Azure service would you host YARP on to serve an unlimited number of custom domains?

I have a multi-tenant SaaS hosted in an Azure App Service/Web App. I'd like to allow my users to just point their custom domains to my IP and thus use their tenants under their own domains; i.e. you set up yourdomain.com to point to my.azure.ip.address and you have your tenant available (there's some backend configuration connecting your tenant to your domain but that's not important here).

To achieve this, I'm looking into putting a reverse proxy in front of the app to act as an entry point. This is to overcome the limitation of App Service where you have to explicitly add and validate custom domains, and only up to 500 is allowed. Formerly I used IIS Application Request Router on a Cloud Service to do this which works well but now I'm looking at newer technologies and thus into YARP.

Given the above requirements, which Azure services would you use to host a YARP instance? I was looking around but in the end it seems to me that still only VMs and Cloud Services have this flexibility. Do you have any other recommendations?

**Comment by samsp-msft** [comment, marked as answer]
I am not an expert on Azure's offerings - you may be better asking the question in their forums etc.
At first glance, I'm wondering if [Azure Container Instances](https://azure.microsoft.com/en-us/services/container-instances/#overview) would be a good choice. Host a linux container running YARP with a static external IP address, and then route from that to the relevant app service app?

**Reply by Piedone** [reply]
That's a great idea! I actually looked at containers before but glanced over the fact that you can also use an IP (and not just an Azure domain). I'll look into this more.

**Reply by Piedone** [reply]
It seems that you can't add a static IP address to an Azure Container Instance. You can do this by creating a VNET and using Azure Application Gateway (see tutorial [here](https://docs.microsoft.com/en-us/azure/container-instances/container-instances-application-gateway) and some more info [here](https://stackoverflow.com/questions/53815807/assign-static-public-address-to-azure-container-instance-deployment)). However, this starts to get quite complex and seems to be an overkill for just having an IP.

So for now I'll try keeping to just use CNAME records where an FQDN can be used too, something that Container Instances do have a static one.
